File: common/src/main/java/net/superricky/tpaplusplus/io/PlayerData.java
package net.superricky.tpaplusplus.io;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.UUID;

public class PlayerData {
    private volatile boolean tpToggle = false;
    private final List<UUID> blockedPlayers = Collections.synchronizedList(new ArrayList<>());

    public boolean getTPToggle() {
        return tpToggle;
    }

    public void setTPToggle(boolean tpToggle) {
        this.tpToggle = tpToggle;
    }

    public List<UUID> getBlockedPlayers() {
        return List.copyOf(blockedPlayers);
    }
}

File: common/src/main/java/net/superricky/tpaplusplus/io/PlayerData.kt
@file:UseSerializers(UUIDSerializer::class)
package net.superricky.tpaplusplus.io

import kotlinx.serialization.Serializable
import kotlinx.serialization.UseSerializers
import java.util.UUID

@Serializable
data class PlayerData(val tpToggle: Boolean = DEFAULT_TP_TOGGLE_STATE, val blockedPlayers: Set<UUID> = setOf()) {
    fun hasBlockedPlayer(player: UUID): Boolean {
        return blockedPlayers.contains(player)
    }

}
